addons/sale_loyalty/tests/test_loyalty.py

```diff
@@ -563,3 +563,57 @@ def test_gift_card_program_without_product(self):
         self._claim_reward(order, giftcard_program)
 
         self.assertEqual(giftcard_program.coupon_count, 0)
+
+    def test_apply_coupon_code_twice(self):
+        """This test ensures that applying a coupon with code twice will:
+            1. Raise an error
+            2. Not delete the coupon
+        """
+
+        # Create a coupon
+        coupon_program = self.env['loyalty.program'].create({
+            'name': 'Coupon Program',
+            'program_type': 'coupons',
+            'trigger': 'with_code',
+            'applies_on': 'current',
+            'reward_ids': [Command.create({
+                    'reward_type': 'discount',
+                    'discount': 10.0,
+                    'discount_mode': 'percent',
+                    'discount_applicability': 'order',
+            })],
+        })
+
+        coupon_partner = self.env['loyalty.card'].create({
+                'program_id': coupon_program.id,
+                'partner_id': self.partner_a.id,
+                'points': 1,
+                'code': '4242',
```

Comment:
is your editor correctly configured? cause you still have double indentation here and below :eyes: